### PR TITLE
Add rerank response serialization

### PR DESCRIPTION
**Pull request by michalkulakowski**
### 🛠 Summary

https://jira.devtools.intel.com/browse/CVS-153726

### 🧪 Checklist

- [x] Unit tests added.
- [ ] The documentation updated.
- [x] Change follows security best practices.
``

### CODE REVIEW

File: src/test/rerank_handler_test.cpp [hunk 1]
```diff
@@ -321,3 +321,87 @@ TEST_F(RerankHandlerTest, InvalidMaxChunksPerDoc) {
...
+    StringBuffer buffer;
+    ASSERT_TRUE(handler.parseRequest().ok());
+    auto status = handler.parseResponse(buffer, scores);
+    EXPECT_TRUE(status.ok());
+    std::string expectedResponse = R"({"results":[{"index":1,"relevance_score":17.2,"text":"second"},{"index":0,"relevance_score":5.36,"text":"first"},{"index":2,"relevance_score":3.0,"text":"third"}]})";
+    EXPECT_STREQ(buffer.GetString(), expectedResponse.c_str());
+}
+
+TEST(RerankHandlerSerializationTest, negativeReturnDocumentsWithDocumentsMap) {  // TODO add support for return documents for documents map
+    std::vector<float> scores = {5.36, 17.21, 3.01, 22.33, 9.4, 22.33};
```
Comment (dtrawins) [inline src/test/rerank_handler_test.cpp:369]: relevance score must be between 0-1

**Review by dtrawins — APPROVED** [review]

File: src/test/rerank_handler_test.cpp [hunk 2]
```diff
@@ -321,3 +321,87 @@ TEST_F(RerankHandlerTest, InvalidMaxChunksPerDoc) {
     auto status = handler.parseRequest();
     EXPECT_EQ(status, absl::InvalidArgumentError("max_chunks_per_doc accepts integer values"));
 }
+
+TEST(RerankHandlerSerializationTest, simplePostivie) {
+    std::vector<float> scores = {5.36, 17.21, 3.01, 22.33, 9.4, 22.33};
```
Comment (dkalinowski) [inline src/test/rerank_handler_test.cpp:326]: check corner values with many decimal places

**Review by dkalinowski — APPROVED** [review]
